httpcli/basic/basic.go

```diff
@@ -9,16 +9,26 @@ import (
 
 // Register registers a DoerFactory for HTTP Basic authentication
 func Register() {
-	httpcli.Register(httpcli.AuthBasic, httpcli.DoerFactory(func(cm httpcli.ConfigMap, c *http.Client) httpcli.Doer {
+	httpcli.Register(httpcli.AuthBasic, httpcli.DoerFactory(func(cm httpcli.ConfigMap, c *http.Client) (doer httpcli.Doer) {
```

Comment:
I'm not sure I see a benefit to giving the return value a name; why not just keep it `return Doer(c, config)`?